src/azure-cli/azure/cli/command_modules/backup/custom.py

```diff
@@ -638,20 +798,20 @@ def restore_disks(cmd, client, resource_group_name, vault_name, container_name,
     if diskslist:
         restore_disk_lun_list = diskslist
 
-    trigger_restore_properties = IaasVMRestoreRequest(create_new_cloud_service=True,
-                                                      recovery_point_id=rp_name,
-                                                      recovery_type='RestoreDisks',
-                                                      region=vault_location,
-                                                      storage_account_id=_storage_account_id,
-                                                      source_resource_id=_source_resource_id,
-                                                      target_resource_group_id=target_rg_id,
-                                                      original_storage_account_option=use_original_storage_account,
-                                                      restore_disk_lun_list=restore_disk_lun_list)
+    trigger_restore_properties = _get_trigger_restore_properties(rp_name, vault_location, _storage_account_id,
+                                                                 _source_resource_id, target_rg_id,
+                                                                 use_original_storage_account, restore_disk_lun_list,
+                                                                 rehydration_duration, rehydration_priority,
+                                                                 rp_list[0].tier_type)
     trigger_restore_request = RestoreRequestResource(properties=trigger_restore_properties)
 
     if use_secondary_region:
         if target_rg_id is None:
             raise RequiredArgumentMissingError("Please provide target resource group using --target-resource-group.")
+
+        if rehydration_priority is not None:
+            raise InvalidArgumentValueError('Cannot have both use_secondary_region and rehydration_priority together.')
```

Comment:
Use MutuallyExclusiveArgumentError class here instead. I think that will make more sense.

Reply:
Let's discuss this tomorrow.